core/vm/evm.go

```diff
@@ -344,9 +332,9 @@ func (evm *EVM) DelegateCall(caller ContractRef, addr common.Address, input []by
 	}
 	var snapshot = evm.StateDB.Snapshot()
 
-	// It is allowed to call precompiles, even via delegatecall
-	if p, isPrecompile := evm.precompile(addr); isPrecompile {
-		ret, gas, err = RunPrecompiledContract(p, input, gas, evm.Config.Tracer)
+	// It is allowed to call precompiles, even via delegatecall, but only for reading
```

Comment:
Have we figured out why only regular CALL opcodes can do state-change, while DELEGATECALL, STATICCALL and CALLCODE became read-only?

Reply:
Not yet. I think this could be a question for Markus or Albert. Will ask in meeting today.